src/data/set/intervals/instances.lean

```diff
@@ -0,0 +1,103 @@
+/-
+Copyright (c) 2022 Stuart Presnell. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stuart Presnell, Eric Wieser, Yaël Dillies
+-/
+import data.set.intervals.basic
+import algebra.order.ring
+import algebra.group_power.order
+
+/-!
+# Structure instances for unit intervals
+
+For suitably structured underlying type `α`, we exhibit the structure of
+the unit intervals from `0` to `1`.
```

Comment:
Can you add a link to the `nonneg.lean` file with a mention that it handles the `Ici 0` instances?